skorch/callbacks/logging.py

```diff
@@ -205,6 +207,76 @@ def on_train_end(self, net, **kwargs):
         if self.close_after_train:
             self.experiment.stop()
 
+class WandbLogger(Callback):
+    """Logs best model and metrics to `Weights & Biases <https://docs.wandb.com/>`_
+
+    "Use this callback to automatically log best trained model and all metrics from
+    your net's history to Weights & Biases after each epoch.
+
+    Examples
+    --------
+    >>> import wandb
+    >>> from skorch.callbacks import WandbLogger
+    >>> wandb_run = wandb.init()
+    >>> wandb.config.update({"learning rate": 1e-3, "batch size": 32})  # optional
+    >>> net = NeuralNet(..., callbacks=[WandbLogger(wandb_run)])
+    >>> net.fit(X, y)
+
+    Parameters
+    ----------
+    wandb_run : wandb.wandb_run.Run
+      wandb Run used to log data.
+
+    save_model : bool (default=True)
+      Saves best trained model.
+
+    keys_ignored : str or list of str (default=None)
+      Key or list of keys that should not be logged to
+      tensorboard. Note that in addition to the keys provided by the
+      user, keys such as those starting with 'event_' or ending on
+      '_best' are ignored by default.
+    """
+
+    # Record if watch has been called previously (even in another instance)
+    _watch_called = False
+
+    def __init__(
+            self,
+            wandb_run,
+            save_model=True,
```

Comment:
We already provide a [checkpoint callback](https://skorch.readthedocs.io/en/stable/callbacks.html#skorch.callbacks.Checkpoint), I think this functionality is redundant.

Reply:
This is to log the trained model to W&B.

Reply:
Interesting. How does that work? In this code, I don't see any interaction with W&B:

```python
        # save best model
        if self.save_model and hist['valid_loss_best']:
            model_path = Path(self.wandb_run.dir) / 'best_model.pth'
            with model_path.open('wb') as model_file:
                net.save_params(f_params=model_file)
```

Is this some code working in the background or is it simply the fact that the model parameters are stored in the `wandb_run_dir`?

Reply:
All files stored in wandb_run.dir are automatically saved.
You can see in my example run on the "files" tab

Reply:
Please leave a comment that states that the files in `wandb_run.dir` is automatically saved in `on_epoch_end`.